rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#define MCAP_IMPLEMENTATION
```

Comment:
This line already appears at <https://github.com/ros-tooling/rosbag2_storage_mcap/blob/main/mcap_vendor/src/main.cpp#L15>, I think including this line in two separate places would have the effect of compiling MCAP into two separate translation units?

Reply:
@jhurliman Nop. If add `#define MCAP_IMPLEMENTATION` here the `mcap` will be used just as includes only from header files directly without linking library at all.

Reply:
@jhurliman To be more accurate I agree that including  `#define MCAP_IMPLEMENTATION` in two separate places would have the effect of compiling MCAP into two separate translation units. 
But these two separate translation units will coexist **independently** and will not overlap.

We need `#define MCAP_IMPLEMENTATION` in `mcap_vendor` package to be able to run unit tests and verify that mcap could be properly used from other packages. It will play like a bridge from ros2 world to the `mcap` C++ implementation. For instance at least provide suitable CMake file which is missing in original `mcap` C++ implementation.

At the same time we need  `#define MCAP_IMPLEMENTATION` in `rosbag2_stoarge_mcap`. 
The main point that using this define in `rosbag2_stoarge_mcap` will force using `mcap` and its dependencies as header only library and `mcap` c++ source code will be compiled directly in `rosbag2_stoarge_mcap` plugin without using binary pieces from `mcap_vendor` package. In some cases it will even positively reflect on performance.

Reply:
@jhurliman Kindly ping to reiterate on review and comments.  
I am kind of blocked on this issue in further development and would like to resolve all requests for changes.

cc: @amacneil

Reply:
I'm not going to be able to add any opinions regarding c++ defines, but I suggest resolving at tooling working group this Friday?

cc @emersonknapp @james-rms @jhurliman